src/main/java/com/thealgorithms/misc/MergeSort.java

```diff
@@ -0,0 +1,86 @@
+class Sort 
+{
+    void merge(int arr[], int left, int middle, int right)
+    {
+        int low = middle - left + 1;                    //size of the left subarray
+        int high = right - middle;                      //size of the right subarray
+ 
+        int L[] = new int[low];                             //create the left and right subarray
+        int R[] = new int[high];
+
+        int i = 0, j = 0;
+ 
+        for (i = 0; i < low; i++)                               //copy elements into left subarray
+        {
+            L[i] = arr[left + i];
+        }
+        for (j = 0; j < high; j++)                              //copy elements into right subarray
+        {
+            R[j] = arr[middle + 1 + j];
+        }
+        
+ 
+        int k = left;                                           //get starting index for sort
+        i = 0;                                             //reset loop variables before performing merge
+        j = 0;
+
+        while (i < low && j < high)                     //merge the left and right subarrays
+        {
+            if (L[i] <= R[j]) 
+            {
+                arr[k] = L[i];
+                i++;
+            }
+            else 
+            {
+                arr[k] = R[j];
+                j++;
+            }
+            k++;
+        }
+ 
+        while (i < low)                             //merge the remaining elements from the left subarray
+        {
+            arr[k] = L[i];
+            i++;
+            k++;
+        }
+ 
+        while (j < high)                           //merge the remaining elements from right subarray
+        {
+            arr[k] = R[j];
+            j++;
+            k++;
+        }
+    }
+ 
+
```

Comment:
helper function that creates the sub cases for sorting

Reply:
What need to change here?